src/parquet/column/reader.cc

```diff
@@ -124,12 +134,16 @@ bool TypedColumnReader<TYPE>::ReadNewPage() {
       // first page with this encoding.
       Encoding::type encoding = page->encoding();
 
-      if (IsDictionaryIndexEncoding(encoding)) {
+      if (IsRleDictionaryIndexEncoding(encoding)) {
         encoding = Encoding::RLE_DICTIONARY;
       }
 
       auto it = decoders_.find(static_cast<int>(encoding));
       if (it != decoders_.end()) {
+        if (encoding == Encoding::RLE_DICTIONARY &&
+            current_decoder_->encoding() != Encoding::RLE_DICTIONARY) {
+            throw ParquetException("Dictionary decoder is not RLE_DICTIONARY.");
+        }
```

Comment:
I don't think this is needed if we have enough tests to feel comfortable decoder was correctly constructed -- alternately this can be made a DCHECK